client/my-sites/plugins/controller-logged-out.js

```diff
@@ -0,0 +1,73 @@
+import { BASE_STALE_TIME } from 'calypso/data/marketplace/constants';
+import {
+	getFetchWPCOMFeaturedPlugins,
+	getFetchWPCOMPlugins,
+} from 'calypso/data/marketplace/use-wpcom-plugins-query';
+import { getFetchWPORGPlugins } from 'calypso/data/marketplace/use-wporg-plugin-query';
+import { requestProductsList } from 'calypso/state/products-list/actions';
+import { getProductsList } from 'calypso/state/products-list/selectors';
+
+function getProps( context ) {
+	const props = {
+		path: context.path,
+		locale: context.lang,
+		tag: '',
+	};
+	return props;
+}
+
+function prefetchPluginsData( queryClient, fetchParams, infinite ) {
+	const queryType = infinite ? 'prefetchInfiniteQuery' : 'prefetchQuery';
+
+	return queryClient[ queryType ]( ...fetchParams, {
+		staleTime: BASE_STALE_TIME,
```

Comment:
This `staleTime` property, I think it could be eliminated from the prefetch call and also from all the `useQuery` hook calls. It could be set once and for all when the `QueryClient` is being created:
```js
queryClient = new QueryClient( {
  defaultOptions: {
    queries: {
      staleTime: BASE_STALE_TIME,
    },
  },
} );
```
Because why would we need to customize it for any individual query?

The `QueryClient` configurations for client and server should be different:
- client: `staleTime` is 0, the default, meaning that the data are stale immediately after fetching. A `useQuery` hook will return the stale data to be displayed in UI, and then refetch immediately to get the fresh data (according to flags like `refetchOnMount`.
- server: `staleTime` is some long time like 2 hours, meaning that the `prefetchQuery` call will fetch only when the data are more than 2 hours old, otherwise it's noop. That how caching is achieved.

Reply:
Fair point. 
`staleTime` removed.